crates/context/src/tx.rs

```diff
@@ -208,3 +208,184 @@ impl Transaction for TxEnv {
         self.gas_priority_fee
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_effective_gas_price_legacy() {
```

Comment:
Lets simplify all these tests and make one with a template.

`fn effective_gas_setup(type: TransactionType, gas_price: u128, priority_fee: Option<u128>) -> u128` 
Base fee can be by default 100

And in test you can do
```
assert_eq!(100, effective_gas_setup(TransactionType::Eip1559, 100, None);
assert_eq!(90, effective_gas_setup(TransactionType::Legacy, 90, None);
```
etc